### DISCUSSION TITLE
How to use GINEConv with the HeteroConv wrapper?

### DISCUSSION THREAD

**Discussion by DogitoErgoSum** [pyg-team/pytorch_geometric#4484] [Q&A]
![image](https://user-images.githubusercontent.com/51344466/163590538-d564df54-6219-4a25-a663-df0db73d7c10.png)
I am using the latter to create the GINEConv convolutions but i cannot find a way to add the neural networks in this loop. How am I supposed to do that?

![image](https://user-images.githubusercontent.com/51344466/163592154-96459dec-f2eb-4432-8d6b-5635989477bc.png)

**Comment by rusty1s** [comment]
You can instantiate the MLP directly inside `GINEConv`:
```python
from torch_geometric import MLP, GINEConv

GINEConv(MLP([-1, hidden_channels, hidden_channels]))
```
Keep in mind that this will only work for node types of same feature dimensionality!

**Reply by DogitoErgoSum** [reply]
If i understood correctly, can i do this?
![image](https://user-images.githubusercontent.com/51344466/163671708-d0d801e6-6f8e-4496-893f-0f55e7d82004.png)

I cant seem to use the MLP class because of the following error

![image](https://user-images.githubusercontent.com/51344466/163671692-96a9d656-e573-4bfc-9807-68669d12996e.png)
![image](https://user-images.githubusercontent.com/51344466/163671697-ecf13cb4-b20e-4270-aeb1-9db8682c424c.png)

**Reply by DogitoErgoSum** [reply]
Ty. 
Relative to what you said there "Keep in mind that this will only work for node types of same feature dimensionality!", if node type A has dim of (1,2) and node type B has dim of (1,10) i cannot use GINEConv? I have to transform both node types to dim(1,X) and then use GINEConv?

**Reply by DogitoErgoSum** [reply]
What i mean is passing all node types to dim=(1,2), as exemplified in blue, and then i am able to use GINEConv (red)?
![image](https://user-images.githubusercontent.com/51344466/163681715-fde89965-9250-474b-bd3d-37930de59f8d.png)

**Reply by DogitoErgoSum** [reply]
Guess i can do this ONLY if use GINConv instead of GINEConv. Thank you for your big patience. How can i acknowledge you or/and the pytorch geometric team in my master´s thesis?

**Reply by rusty1s** [reply]
This looks correct to me. Glad that it is working. Citing our paper is all we need :)

**Comment by rusty1s** [comment]
```
pip install class-resolver
```
Should fix this :)
